controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -162,12 +168,410 @@ import {
 } from '../util.js';
 import { FederatedGraphSchemaUpdate, OrganizationWebhookService } from '../webhooks/OrganizationWebhookService.js';
 import { AuditLogRepository } from '../repositories/AuditLogRepository.js';
+import { DefaultNamespace, NamespaceRepository } from '../repositories/NamespaceRepository.js';
+import { PublicError } from '../errors/errors.js';
 
 export default function (opts: RouterOptions): Partial<ServiceImpl<typeof PlatformService>> {
   return {
     /*
     Mutations
     */
+
+    createNamespace: (req, ctx) => {
+      const logger = opts.logger.child({
+        service: ctx.service.typeName,
+        method: ctx.method.name,
+      });
+
+      return handleError<PlainMessage<CreateNamespaceResponse>>(logger, async () => {
+        const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
+        const namespaceRepo = new NamespaceRepository(opts.db, authContext.organizationId);
+        const auditLogRepo = new AuditLogRepository(opts.db);
+
+        const exists = await namespaceRepo.byName(req.name);
+        if (exists) {
+          return {
+            response: {
+              code: EnumStatusCode.ERR_ALREADY_EXISTS,
+              details: 'The namespace already exists.',
+            },
+          };
+        }
+
+        const ns = await namespaceRepo.create({
+          name: req.name,
+          createdBy: authContext.userId,
+        });
+
+        await auditLogRepo.addAuditLog({
+          organizationId: authContext.organizationId,
+          auditAction: 'namespace.created',
+          action: 'created',
+          actorId: authContext.userId,
+          auditableType: 'namespace',
+          auditableDisplayName: ns.name,
+          actorDisplayName: authContext.userDisplayName,
+          actorType: authContext.auth === 'api_key' ? 'api_key' : 'user',
+        });
+
+        return {
+          response: {
+            code: EnumStatusCode.OK,
+          },
+        };
+      });
+    },
+
+    deleteNamespace: (req, ctx) => {
```

Comment:
Deleting a namespace should only be possible by the creator or admin. What's the impact of deleting a namespace? I'd expect to delete all the resources in it.

Reply:
Done

Reply:
Fixed